File: 144.BinaryTreePreOrder.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def preOrder(self, root, nodes):
        if root == None:
            return 
        nodes.append(root.val)
        self.preOrder(root.left, nodes)
        self.preOrder(root.right, nodes)


    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        nodes = []
        self.preOrder(root, nodes)
        return nodes

File: 144.BinaryTreePreOrder.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    List<Integer> nodes = new ArrayList<>();

    void preOrder(TreeNode root) {
        if(root == null) {
            return;
        }
        nodes.add(root.val);
        preOrder(root.left);
        preOrder(root.right);
    }

    public List<Integer> preorderTraversal(TreeNode root) {
        preOrder(root);
        return nodes;
    }
}
